tests/unit/s2n_connection_test.c

```diff
@@ -239,37 +239,27 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_get_selected_signature_algorithm(conn, &output));
         EXPECT_EQUAL(S2N_TLS_SIGNATURE_ANONYMOUS, output);
 
-        s2n_signature_algorithm inputs[] = { S2N_SIGNATURE_ANONYMOUS, S2N_SIGNATURE_RSA, 
-                                             S2N_SIGNATURE_ECDSA, S2N_SIGNATURE_RSA_PSS_RSAE, 
-                                             S2N_SIGNATURE_RSA_PSS_PSS };
-        s2n_tls_signature_algorithm expected_output[] = { S2N_TLS_SIGNATURE_ANONYMOUS, S2N_TLS_SIGNATURE_RSA, 
-                                                          S2N_TLS_SIGNATURE_ECDSA, S2N_TLS_SIGNATURE_RSA_PSS_RSAE, 
-                                                          S2N_TLS_SIGNATURE_RSA_PSS_PSS };
-        EXPECT_EQUAL(s2n_array_len(expected_output), s2n_array_len(inputs));
-
+        s2n_tls_signature_algorithm expected_output[] = {
+            [ S2N_SIGNATURE_ANONYMOUS ] = S2N_TLS_SIGNATURE_ANONYMOUS,
+            [ S2N_SIGNATURE_RSA ] = S2N_TLS_SIGNATURE_RSA,
+            [ S2N_SIGNATURE_ECDSA ] = S2N_TLS_SIGNATURE_ECDSA,
+            [ S2N_SIGNATURE_RSA_PSS_RSAE ] = S2N_TLS_SIGNATURE_RSA_PSS_RSAE,
+            [ S2N_SIGNATURE_RSA_PSS_PSS ] = S2N_TLS_SIGNATURE_RSA_PSS_PSS,
+        };
+        
         for (size_t i = 0; i <= UINT16_MAX; i++) {
-            if (i < s2n_array_len(inputs) && s2n_is_valid_signature_algorithm(inputs[i])) {
-                conn->secure.client_cert_sig_scheme.sig_alg = inputs[i];
+            conn->secure.client_cert_sig_scheme.sig_alg = i;
+            conn->secure.conn_sig_scheme.sig_alg = i;
+            if (s2n_is_valid_signature_algorithm(i)) {
```

Comment:
Much better, but do you still need s2n_is_valid_signature_algorithm? Can you just check (i < s2n_array_len(expected_output)? If a value isn't set, it'll be 0 / S2N_TLS_SIGNATURE_ANONYMOUS.

Reply:
Yes, we do need this function because the value `i = 2` is not valid input.  The valid inputs are 0, 1, 3, 224 and 225: 
```
typedef enum {
    S2N_TLS_SIGNATURE_ANONYMOUS = 0,
    S2N_TLS_SIGNATURE_RSA = 1,
    S2N_TLS_SIGNATURE_ECDSA = 3,

    /* Use Private Range for RSA PSS since it's not defined there */
    S2N_TLS_SIGNATURE_RSA_PSS_RSAE = 224,
    S2N_TLS_SIGNATURE_RSA_PSS_PSS
} s2n_tls_signature_algorithm;
```

Reply:
What would be the value of expected_output[2] then?

Reply:
It will be 0 as expected, updated in new revision.